tests/unit/test_index_contents_parse.py

```diff
@@ -0,0 +1,426 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Unit tests for index module _get_contents_parsed_items function."""
+
+# Need access to protected functions for testing
+# pylint: disable=protected-access
+
+from itertools import chain
+
+import pytest
+
+from src import exceptions, index, types_
+
+from .. import factories
+from .helpers import assert_substrings_in_string
+
+
+def _test__get_contents_parsed_items_invalid_parameters():
+    """Generate parameters for the test__get_contents_parsed_items_invalid test.
+
+    Returns:
+        The tests.
+    """
+    return [
+        pytest.param(
+            f"""# Contents
+{(line := ' - [title 1](value 1)')}""",
+            (line,),
+            id="first item has single leading space",
+        ),
+        pytest.param(
+            f"""# Contents
+{(line := '  - [title 1](value 1)')}""",
+            (line,),
+            id="first item has multiple leading space",
+        ),
+        pytest.param(
+            f"""# Contents
+{(line := '- [title 1](value 1)other')}""",
+            (line,),
+            id="first item trailing non-whitespace",
+        ),
+        pytest.param(
+            f"""# Contents
+{(line := 'malformed')}""",
+            (line,),
+            id="single malformed line",
+        ),
+        pytest.param(
+            f"""# Contents
+- [title 1](value 1)
+{(line := 'malformed')}""",
+            (line,),
+            id="multiple lines single malformed line second",
+        ),
+        pytest.param(
+            f"""# Contents
+{(line := 'malformed')}
+- [title 1](value 1)""",
+            (line,),
+            id="multiple lines single malformed line first",
+        ),
+        pytest.param(
+            f"""# Contents
+{(line := 'malformed 1')}
+malformed 2""",
+            (line,),
+            id="multiple malformed lines",
+        ),
+    ]
```

Comment:
We should perhaps try something like:
```
# Contents
- [title 1](value 1)
  - [title 1.1](value 1.1)
    - [title 1.1.1](value 1.1.1)
   - [title 1.2](valie 1.2)
 ```

Reply:
We should also perhaps try:
`-`
and
`- [title 1]`
and
`- [title 1(value1)`
and
`- [title 1](value 1`
and
`- [title 1] (value 1)`

Reply:
Yeap the first one is not a good one